### PR TITLE
[bitnami/thanos] feat: allow disabling emptydir for byo volume template

### PR DESCRIPTION
**Pull request by aslafy-z**
### Description of the change

Allow disabling the emptyDir mount when builtin persistence is disabled. 

### Benefits

This allows the use of custom volumes, ephemeral volumes or hostpath for eg.

### Applicable issues

- related https://github.com/bitnami/charts/issues/10773

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/thanos/values.yaml [hunk 1]
```diff
@@ -2173,6 +2173,9 @@ compactor:
     ## @param compactor.persistence.enabled Enable data persistence using PVC(s) on Thanos Compactor pods
     ##
     enabled: true
+    ## @param compactor.persistence.defaultsToEmptyDir Defaults to emptyDir if persistence is disabled.
+    ##
+    defaultsToEmptyDir: true
```
Comment (aslafy-z) [inline bitnami/thanos/values.yaml:2178]:
```suggestion
    defaultEmptyDir: true
```

**Comment by carrodher** [comment]
Thank you for initiating this pull request. We appreciate your effort. Just a friendly reminder that it's important to sign your commits. Adding your signature certifies that you either authored the patch or have the necessary rights to contribute the changes. You can find detailed information on how to do this in the [“Sign your work” section of our contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).

Feel free to reach out if you have any questions or need assistance with the signing process.

**Comment by aslafy-z** [comment]
@carrodher I did the required changes :)

**Review by Mauraza — APPROVED** [review]
Hi @aslafy-z,

Thanks for the contribution!! LGTM!!

**Pull request merged**